src/hal/components/sampler.c

```diff
@@ -115,7 +115,7 @@ int rtapi_app_main(void)
     samplers = hal_malloc(MAX_SAMPLERS * sizeof(sampler_t));
     /* validate config info */
     for ( n = 0 ; n < MAX_SAMPLERS ; n++ ) {
-	if (( cfg[n] == NULL ) || ( *cfg == '\0' ) || ( depth[n] <= 0 )) {
+	if (( cfg[n] == NULL ) || ( *cfg[n] == '\0' ) || ( depth[n] <= 0 )) {
```

Comment:
now they both do the same thing.

Reply:
I am not sure they do, cfg is an array of pointers, to we are checking if each element is a null pointer, and if it points to an empty string.

Reply:
Maybe you need `*(cfg[n]) == '\0'`, with added parens so there's no question of operator precedence between * and []?